### PR TITLE
Add undo feature

### PR DESCRIPTION
**Pull request by bennyE31**
Add feature to track previous moves and undo upon user input with 'u'.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -74,6 +75,8 @@ static int   g_sudokuCount = 1;				/* in case of -n we can the numbers of sudoku
 static bool  g_outIsPDF;
 static DIFFICULTY g_level = D_EASY;
 static WINDOW *grid, *infobox, *status;
+static move_t	g_undo_stack[UNDO_STACK_SIZE];	/* Stack of previous moves */
+static int   g_undo_stack_ptr = 0;
```
Comment (jubalh) [inline src/main.c]: Somehow I dont like the `_ptr` without it being a `*`. Maybe `_index` would be a better description? item?

File: src/main.c [hunk 2]
```diff
@@ -337,12 +340,34 @@ static void init_windows(void)
 	wprintw(infobox, _(" r - Redraw\n"));
 	wprintw(infobox, _(" S - Solve puzzle\n"));
 	wprintw(infobox, _(" x - Delete number\n"));
+	wprintw(infobox, _(" u - Undo previous move\n"));
```
Comment (jubalh) [inline src/main.c]: We are not undoing moving in the grid but undoing insertions/actions

**Review by jubalh — CHANGES_REQUESTED** [review]
Thanks a lot for your PR!
I left a couple of comments.

And one more thing:
I go to a field and insert `1` I go to another and insert `2`. Now I go to the field where I insert `1` and press delete. When now pressing `u` it is not adding `1` back.
This might be confusing.

File: src/main.c (hunk 1 above)
Comment (bennyE31) [reply src/main.c]: makes sense, I like index as well. updated

File: src/main.c (hunk 2 above)
Comment (bennyE31) [reply src/main.c]: Yeah I was thinking move in the game sense like a chess "move", not a traversing the board "move." I like action as it covers both insertion and deletion

**Comment by bennyE31** [comment]
> Thanks a lot for your PR! I left a couple of comments.
> 
> And one more thing: I go to a field and insert `1` I go to another and insert `2`. Now I go to the field where I insert `1` and press delete. When now pressing `u` it is not adding `1` back. This might be confusing.

Yep I had just not thought of deletion as an undoable action, which it easily can and should be. I've added that in now and it should work as you expected

**Review by jubalh — APPROVED** [review]
Sorry for the long waiting time! It looks good now! I don't see a reason to have `move_t` in the header though. Will merge anyways :)

**Pull request merged**